rep_cls.py

```diff
@@ -0,0 +1,398 @@
+            ValueError: if numbers in pca_dimensions are not all positive
+                integers.
+            TypeError: if img_format is not an ImgFormat object.
+        """
+
+        # Read embeddings and labels from file and store in self.embeddings and
+        # self.labels
+        self.embeddings = None
+        self.embedding_file = embedding_file
+        self.labels = None
+        self.label_file = label_file
+
+        # Set img_dir and names of all images
+        # Get mapping from label to image for naive distance calculation
+        self._img_names = None
+        self._label_img_map = None
+        self.img_dir = img_dir
+
+        # Assertion that # of embeddings, labels match
+        if len(self.embeddings) != len(self.labels):
+            raise ValueError("Embeddings and labels should have the same number"
+                             " of entries.")
+        # Assertion that there are not less images than embeddings
+        if len([name for name in self._img_names if
+                os.path.isfile(name)]) < len(self.labels):
+            raise ValueError("Not enough image files, make sure img_dir "
+                             "contains the whole dataset.")
+
+        # Set number of candidates
+        self.n_candidates = n_candidates
+
+        # Get low dimensional embeddings based on PCA
+        self.pca_dimensions = pca_dimensions
+
+        # Set image format
+        self.img_format = img_format
+
+        # Primary filter setup
+        self.primary_distance_type = primary_distance_type
+        # Secondary filter setup
+        self.secondary_distance_type = secondary_distance_type
+        self.secondary_filter_threshold = secondary_filter_threshold
+
+    @property
+    def embedding_file(self):
+        """
+        Returns:
+            self._embedding_file: Str, path to the embedding (_vec) .tsv file.
+        """
+        return self._embedding_file
+
+    @property
+    def label_file(self):
+        """
+        Returns:
+            self._label_file: Str, path to the label (_meta) .tsv file.
+        """
+        return self._label_file
+
+    @property
+    def img_dir(self):
+        """
+        Returns:
+            self._img_dir: Str, location of the image folder generated by
+                VisualGenerator.
+        """
+        return self._img_dir
+
+    @property
+    def n_candidates(self):
+        """
+        Returns:
+            self._n_candidates: Int, number of candidates to choose from.
+        """
+        return self._n_candidates
+
+    @property
+    def pca_dimensions(self):
+        """
+        Returns:
+            self._pca_dimensions: List of Int, number of principal components.
+        """
+        return self._pca_dimensions
+
+    @property
+    def img_format(self):
+        """
+        Returns:
+            self._img_format: ImgFormat, format of specified images.
+        """
+        return self._img_format
+
+    @embedding_file.setter
```

Comment:
I think a setter should not perform way too much tasks than simple validation or setting. I prefer you, instead of make these as setter for embedding_file, rename them as  read_embedding_file() read_label_file(), read_images(), create_pca_components(), etc. n_candidates or img_format are reasonable setters since it only did simple validation.